pkg/content/content.go

```diff
@@ -0,0 +1,161 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package content
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/remotes"
+	"github.com/containerd/containerd/snapshots"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/goharbor/acceleration-service/pkg/config"
+	"github.com/goharbor/acceleration-service/pkg/remote"
+)
+
+var logger = logrus.WithField("module", "content")
+
+// ContentProvider provides necessary image utils, image content
+// store for image conversion.
+type ContentProvider interface {
+	// Pull pulls source image from remote registry by specified reference.
+	Pull(ctx context.Context, ref string) error
+	// Push pushes target image to remote registry by specified reference,
+	// the desc parameter represents the manifest of targe image.
+	Push(ctx context.Context, desc ocispec.Descriptor, ref string) error
+
+	// Write writes a blob to containerd content store, the blob can be an
+	// image manifest, index, layer or config.
+	Write(ctx context.Context, desc ocispec.Descriptor, reader io.Reader, labels map[string]string) error
+	// Read reads a blob from containerd content store.
+	Read(ctx context.Context, desc ocispec.Descriptor) ([]byte, error)
```

Comment:
Now that we have `ContentProvider.ContentStore()` API, so could we remove `Write` and `Read` APIs and let the caller to use `ContentStore` instead?